src/Hosting.Services.Web/Middlewares/UserIdentity/IdentityProviders/EmailBasedUserIdentityProvider.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
+{
+	internal class EmailBasedUserIdentityProvider : IUserIdentityProvider
+	{
+        private const int MaxBytesPerChar = 4; // UTF-8
+		public int MaxBytesInIdentity { get; } = 256 * MaxBytesPerChar; // maximum email address length plus max bytes per char
+
+        private class UserEmail
+        {
+            public string Email { get; set; } = string.Empty;
+        }
+
+		public async Task<(bool success, int bytesWritten)> TryWriteBytesAsync(HttpContext context, Memory<byte> memory)
+		{
+			int bytesWritten = -1;
+            try{
+                context.Request.Body.Seek(0, SeekOrigin.Begin);
+                UserEmail? email;
+                using (StreamReader reader = new(context.Request.Body)){
+                    email = await JsonSerializer.DeserializeAsync<UserEmail>(reader.BaseStream);
+                }
+
+                bool success = email != null;
+                if (email != null)
```

Comment:
>email != null [](http://example.com/codeflow?start=20&length=13)

If `success` would be slightly cleaner, since you've already performed the check

Reply:
I know it looks weird, but it is needed for line `foreach (char c in email.Email)`, otherwise a "possibly null" error will be reported.

Reply:
I mean that `email != null` and `success` are the same. But maybe the current design is better as it makes it clearer why it's being done – not that it is successful, but to prevent a null error